File: TypeScript/leetcode/27-RemoveElement.ts
import { assertEquals } from "https://deno.land/std@0.192.0/testing/asserts.ts";

function removeElement(nums: number[], val: number): number {
  let k = 0;

  for (let i = 0; i < nums.length; i++) {
    if (nums[i] != val) {
      nums[k] = nums[i];
      k++;
    }
  }
  return k;
}

Deno.test("TestRemoveElement", () => {
  const nums = [3, 2, 2, 3];
  const val = 3;
  const k = removeElement(nums, val);
  const expect = 2;
  const expectedNums = [2, 2];
  assertEquals(k, expect);
  for (let i = 0; i < k; i++) {
    assertEquals(nums[i], expectedNums[i]);
  }
});

File: Python/algoritms/leetCode/27-remove_element.py
import unittest


def remove_element(nums: list[int], val: int) -> int:
    k = 0

    for i in range(len(nums)):
        if nums[i] != val:
            nums[k] = nums[i]
            k += 1
    return k


class Test(unittest.TestCase):
    def test_remove_element(self) -> None:
        nums = [3, 2, 2, 3]
        val = 3
        k = remove_element(nums, val)
        print(nums)
        expect = 2
        expectedNums = [2, 2]
        self.assertEqual(k, expect)

        for i in range(k):
            self.assertEqual(nums[i], expectedNums[i])


if __name__ == "__main__":
    unittest.main()
